Classes/URKArchive.mm

```diff
@@ -331,7 +403,32 @@ - (nullable NSURL *)firstVolumeURL {
 
 - (nullable NSArray<NSString*> *)listVolumePaths:(NSError **)error
 {
-    return @[];
+    __block NSMutableArray<NSString*> *volumePaths = [NSMutableArray new];
+    
+    URKArchive * dummy = self;
+    NSURL * firstVolumeURL = [self firstVolumeURL];
+    if (firstVolumeURL != self.fileURL)
+    {
+        NSError *firstVolumeURLError = nil;
+        
+        dummy = [self initWithURL:firstVolumeURL error:&firstVolumeURLError];
```

Comment:
In `listVolumePaths` we need to return all the volumes available. If, for example, I've created the `URKArchive` instance with volume 3 of 4, when checking `listFileInfo` it will only return 3 and 4. To get 1-3, self.fileURL need to be the volume 1. So this pretty piece of code did that, changing self.fileURL to `firstVolumeURL`, when needed.

That's why I told you about the overload with `fileURL`, so I can use some methods with other files than `self.fileURL`.